converter/grpc_interceptor.go

```diff
@@ -0,0 +1,268 @@
+	case *workflowservice.StartWorkflowExecutionRequest:
+		return s.encodePayloads(r.Input)
+	case *workflowservice.SignalWorkflowExecutionRequest:
+		return s.encodePayloads(r.Input)
+	case *workflowservice.SignalWithStartWorkflowExecutionRequest:
+		return s.encodePayloads(r.Input, r.SignalInput)
+	case *workflowservice.TerminateWorkflowExecutionRequest:
+		return s.encodePayloads(r.Details)
+	case *workflowservice.RespondActivityTaskCompletedRequest:
+		return s.encodePayloads(r.Result)
+	case *workflowservice.RespondActivityTaskCompletedByIdRequest:
+		return s.encodePayloads(r.Result)
+	case *workflowservice.RespondActivityTaskCanceledRequest:
+		return s.encodePayloads(r.Details)
+	case *workflowservice.RespondActivityTaskCanceledByIdRequest:
+		return s.encodePayloads(r.Details)
+	case *workflowservice.RespondActivityTaskFailedRequest:
+		return s.encodeFailure(r.Failure)
+	case *workflowservice.RespondActivityTaskFailedByIdRequest:
+		return s.encodeFailure(r.Failure)
+	case *workflowservice.RespondWorkflowTaskCompletedRequest:
+		return s.processCommands(r.Commands)
+	case *workflowservice.RespondWorkflowTaskFailedRequest:
+		return s.encodeFailure(r.Failure)
+	case *workflowservice.RecordActivityTaskHeartbeatRequest:
+		return s.encodePayloads(r.Details)
+	case *workflowservice.RecordActivityTaskHeartbeatByIdRequest:
+		return s.encodePayloads(r.Details)
+	}
+
+	return nil
+}
+
+func (s *serviceInterceptor) processResponse(response interface{}) error {
+	switch r := response.(type) {
+	case *workflowservice.GetWorkflowExecutionHistoryResponse:
+		return s.processEvents(r.History.GetEvents())
+	case *workflowservice.PollWorkflowTaskQueueResponse:
+		return s.processEvents(r.History.GetEvents())
+	case *workflowservice.PollActivityTaskQueueResponse:
+		return s.decodePayloads(r.Input, r.HeartbeatDetails)
+	}
+
+	return nil
+}
+
+func (s *serviceInterceptor) encodeFailure(failure *failure.Failure) error {
+	if failure == nil {
+		return nil
+	}
+
+	err := s.encodePayloads(
+		failure.GetApplicationFailureInfo().GetDetails(),
+		failure.GetCanceledFailureInfo().GetDetails(),
+		failure.GetResetWorkflowFailureInfo().GetLastHeartbeatDetails(),
+	)
+	if err != nil {
+		return err
+	}
+
+	return s.encodeFailure(failure.GetCause())
+}
+
+func (s *serviceInterceptor) decodeFailure(failure *failure.Failure) error {
+	if failure == nil {
+		return nil
+	}
+
+	err := s.decodePayloads(
+		failure.GetApplicationFailureInfo().GetDetails(),
+		failure.GetCanceledFailureInfo().GetDetails(),
+		failure.GetResetWorkflowFailureInfo().GetLastHeartbeatDetails(),
+	)
+	if err != nil {
+		return err
+	}
+
+	return s.encodeFailure(failure.GetCause())
+}
+
+func (s *serviceInterceptor) processCommands(commands []*command.Command) error {
+	var err error
+	for _, c := range commands {
+		switch c.CommandType {
+		case enumspb.COMMAND_TYPE_SCHEDULE_ACTIVITY_TASK:
+			err = s.encodePayloads(c.GetScheduleActivityTaskCommandAttributes().Input)
```

Comment:
Might want to use `GetInput()` on these just in case for whatever reason the attributes are nil